core/src/test/scala/org/apache/spark/scheduler/FakeTask.scala

```diff
@@ -36,34 +36,74 @@ class FakeTask(
   override def preferredLocations: Seq[TaskLocation] = prefLocs
 }
 
+class FakeLongTasks(
```

Comment:
Can we define it in the `TaskSetManagerSuite` and avoid touching this file? This looks like a very special case.

Reply:
Just to keep as a utility for future case and as the `FakeTask.create` is in a separate file i have updated this in `FakeTask.scala`

Reply:
If you insist, can move the class and creation logic to 'TaskSetManagerSuite' but it will result in some code duplication with `FakeTask.createTasks`, should i move it.?

Reply:
I don't think there will be many duplications: we just create a task array and put it in TaskSet
```
Array.tabulate[Task[_]](numTasks) { i =>
  new FakeLongTasks(0, i)
}
...

class FakeLongTasks(partitionId: Int)
  extends Task[Int](0, 0, partitionId, new Properties,
    SparkEnv.get.closureSerializer.newInstance().serialize(TaskMetrics.registered).array(), false) ...
```

Reply:
I have updated the PR as per your suggestion